### DISCUSSION TITLE
Send WOL packet on container start

### DISCUSSION THREAD

**Discussion by trentnbauer** [seriousm4x/UpSnap#381] [Feature requests]
I would love to see the option to send a WOL packet on container start. It would be great to set a delay, as well.
My use case is
- UpSnap installed on Linguine, which isn't protected by battery backup
- UpSnap is used to wake Macaroni (proxmox host) and Fettuccine (synology NAS)
![image](https://github.com/seriousm4x/UpSnap/assets/45578026/8122e754-7ab9-4a35-ae4a-cb14b6d9b81b)

My goal is for power outages to follow this workflow,
1. Power outage occurs, Linguine loses power
2. UPS software (hosted on Macaroni) waits until 10 minute run time
3. UPS software powers down servers
4. UPS continues to run with close to 0 load. UPS either runs flat or lasts the outage
5. Power returns, Linguine boots
6. Linguini sends WOL packet to servers

'Power on with AC' is an option here, but I think it is a poor one as it allows the server to boot
- when I don't want it too (eg doing maintenance)
- during a very short return of power
- only powers the server on if the UPS runs flat

(yes, my servers are named after pasta)

**Comment by seriousm4x** [comment]
Hey, I understand the use case but there might already be a working solution for you by using the wake cron.

When editing a device you could use set the wake cron to `*/1 * * * *` and the server would boot up after AC loss. This will send a wake request once every minute. If the server is already online, the request will be skipped. It's not instant, but you could also use `@every 5s` for the wake cron to reduce the time. (I don't recommend using lower values.)

![image](https://github.com/seriousm4x/UpSnap/assets/23456686/d392464a-f60b-44e3-af0a-149f26eb3ba2)

If that doesn't work for you, you could create a simple shell script on your Linguine server using the [API](https://github.com/seriousm4x/UpSnap/wiki/Rest-API) to wake the other server.

First authenticate to get a bearer token. Then send a wake request to `/api/upsnap/wake/:id` with the bearer token set in the header. (You can see the `:id` in the url when editing a device)

**Reply by trentnbauer** [reply]
I'm not sure if the */1 cron will be an option for my usecase - this would wake the server if I had purposely shut it down, correct?

I'll have a look into the API :)

**Reply by seriousm4x** [reply]
> I'm not sure if the */1 cron will be an option for my usecase - this would wake the server if I had purposely shut it down, correct?

Yes, this will always power the server on. For implying your own logic, I think the api is your best option.
